frontend/model/captureLogs.js

```diff
@@ -0,0 +1,186 @@
+
+  console.debug('Starting to capture logs of type:', appLogsFilter)
+
+  // Override the console to start capturing the logs
+  if (!isConsoleOveride) {
+    // avoid duplicated captures, in case captureLogsStart
+    // is called multiple times. (ex: login twice in the same visit)
+    isConsoleOveride = true
+
+    const _cError = console.error
+    const _cWarn = console.warn
+    const _cDebug = console.debug
+    const guardCapture = (type, args) => {
+      if (isCapturing && appLogsFilter.includes(type)) {
+        captureLog(type, ...args)
+      }
+    }
+
+    console.error = function () {
+      _cError.apply(console, arguments)
+      guardCapture('error', arguments)
+    }
+    console.warn = function () {
+      _cWarn.apply(console, arguments)
+      guardCapture('warn', arguments)
+    }
+    console.debug = function () {
+      _cDebug.apply(console, arguments)
+      guardCapture('debug', arguments)
+    }
+  }
+
+  // // REMOVE THIS: It overrides all types,
+  // // (including log and info). It's not needed.
+  // window.console = new Proxy(window.console, {
+  //   get: (obj, type) => (...args) => {
+  //     obj[type](...args)
+  //     if (isCapturing && appLogsFilter.includes(type)) {
+  //       captureLog(type, ...args)
+  //     }
+  //   }
+  // })
+}
+
+export function captureLogsPause () {
+  isCapturing = false
+  sbp('okTurtles.events/off', SET_APP_LOGS_FILTER)
+}
+
+// Util to download *all* stored logs so far
+export function downloadLogs (filename, elLink) {
+  const logsArr = []
+  let lastEntry = localStorage.getItem('giConsole/lastEntry')
+  do {
+    const entry = localStorage.getItem(`giConsole/${lastEntry}`)
+    const entryParsed = JSON.parse(entry) || {}
+    lastEntry = entryParsed.prev
+    logsArr.push(entry)
+  } while (lastEntry)
+  const data = `/*
+. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 
+
+GROUP INCOME - Application Logs
+
+Attach this file when reporting a problem:
+  - Github: https://github.com/okTurtles/group-income-simple/issues  
+
+. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
+*/
+
+var userAgent = ${navigator.userAgent}
+var appLogs = [${logsArr}]
+`
+
+  const file = new Blob([data], { type: 'text/plain' })
+
+  if (window.navigator.msSaveOrOpenBlob) {
+    // IE10+ and Edge
+    console.log('download using MS API')
+    window.navigator.msSaveOrOpenBlob(file, filename)
+  } else {
+    console.log('download using URL obj')
+
+    const url = URL.createObjectURL(file)
+    elLink.href = url
+    elLink.download = filename
+    elLink.click()
+    setTimeout(() => {
+      elLink.href = '#'
+      window.URL.revokeObjectURL(url)
+    }, 0)
+  }
+}
+
+export function clearLogs () {
+  let i = localStorage.length
+  while (i--) {
+    const key = localStorage.key(i)
+    if (/giConsole/.test(key)) {
+      localStorage.removeItem(key)
+    }
+  }
+  lastEntry = ''
+  entriesCount = 0
```

Comment:
Remember, these variables should be kept in sync with localStorage, so if you reset them here you need to reset both of them in localStorage as well.